### ISSUE TITLE
Enhancement: Address XXX lines in top level files

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/temp-test-ioccc#2008]
In various top level files, we have **XXX** lines indicating that some better text needs to replace the XXX note.

# Critical path TODO

* [x] `/faq.md#submit` FAQ 0.0 - How to register

The text "XXX - instructions TBD - XXX" related to how to register needs to better written before the next news update to the official IOCCC is posted.

While the exact process of how to "Register for the IOCCC" has not been determined, a much better note than "XXX ... XXX" need to be in place informing the general public the instructions are not yet ready.  Of course this instructions needs to have a "XXX" marker added to them so that when the instructions a ready, the full instructions may be added.

Our point is that "XXX ... XXX"  is **NOT a good state** for people to see.  A better "XXX" comment needs to be in place indicating that the "how to register" instructions are not ready and they should check back later (along some "XXX" in the note for even later resolving.

* [x] `/faq.md#submit` FAQ 0.0 - How to Upload your entry

The text "XXX - instructions TBD - XXX" related to how to register needs to better written before the next news update to the official IOCCC is posted.

While the exact process of how to "Upload your entry to the IOCCC submit server" has not been determined, a much better note than "XXX ... XXX" need to be in place informing the general public the instructions are not yet ready.  Of course this instructions needs to have a "XXX" marker added to them so that when the instructions a ready, the full instructions may be added.

Our point is that "XXX ... XXX"  is **NOT a good state** for people to see.  A better "XXX" comment needs to be in place indicating that the "Upload your entry to the IOCCC submit server" instructions are not ready and they should check back later (along some "XXX" in the note for even later resolving.

# Change of state TODO

* [x] When the all of the above "Critical path TODO" is done, remove the **((top priority))** Label from this issue.

# General TODO list

* [x] `/faq.md#fix_web_site` FAQ 5.4

When the complete set of tools referenced in issue #4 is written, then the "FAQ 5.4: How may I submit a fix to the IOCCC web site?" entry needs to be updated to help people understand what may be modified in a pull request and what should **NOT be modified**.

Issue #1933 also impacts this TODO item, and so probably that issue should be mostly completed if not resolved before this TOOD item might be ready to be completed.

**Comment by xexyl** [comment]
> In various top level files, we have **XXX** lines indicating that some better text needs to replace the XXX note.
> 
> # Critical path TODO
> * [ ]  `/faq.md#submit` FAQ 0.0 - How to register
> 
> The text "XXX - instructions TBD - XXX" related to how to register needs to better written before the next news update to the official IOCCC is posted.

Is this possible without the new server being set up (the server to submit etc.)?

> 
> While the exact process of how to "Register for the IOCCC" has not been determined, a much better note than "XXX ... XXX" need to be in place informing the general public the instructions are not yet ready. Of course this instructions needs to have a "XXX" marker added to them so that when the instructions a ready, the full instructions may be added.
> 

Hmm .. okay so you see my point. But I'm still not sure how to make it a better one without knowing the details?

> Our point is that "XXX ... XXX" is **NOT a good state** for people to see. A better "XXX" comment needs to be in place indicating that the "how to register" instructions are not ready and they should check back later (along some "XXX" in the note for even later resolving.

Agreed.

> 
> * [ ]  `/faq.md#submit` FAQ 0.0 - How to Upload your entry
> 
> The text "XXX - instructions TBD - XXX" related to how to register needs to better written before the next news update to the official IOCCC is posted.

Same question as above. It might also be good to also reference the FAQ (was there one?) in the other repo?

> While the exact process of how to "Upload your entry to the IOCCC submit server" has not been determined, a much better note than "XXX ... XXX" need to be in place informing the general public the instructions are not yet ready. Of course this instructions needs to have a "XXX" marker added to them so that when the instructions a ready, the full instructions may be added.
> 

Okay so informing the people it's not yet been determined .. but what else?

> Our point is that "XXX ... XXX" is **NOT a good state** for people to see. A better "XXX" comment needs to be in place indicating that the "Upload your entry to the IOCCC submit server" instructions are not ready and they should check back later (along some "XXX" in the note for even later resolving.
> 
Agreed there as above.

> # General TODO list
> * [ ]  `/README.md`
> 
> Remove from the top line down to an including the line "## What follows is the actual README.md file .." when the **great fork merge is ready** to be performed.
> 
This seems like it should be in a final todo list?

> * [ ]  `/faq.md#fix_web_site` FAQ 5.4
> 
> When the complete set of tools referenced in issue #4 is written, then the "FAQ 5.4: How may I submit a fix to the IOCCC web site?" entry needs to be updated to help people understand what may be modified in a pull request and what should **NOT be modified**.

I'm not sure what this means yet but it might be I'm not awake enough. I'm happy to be awake in a way though as I get a rare treat: listening to the rain!

> 
> Issue #1933 also impacts this TODO item, and so probably that issue should be mostly completed if not resolved before this TOOD item might be ready to be completed.
> 
Okay. Maybe (just a random idea) a prerequisite todo list could be here? I don't know: just a thought that might (or more likely does not) have some merit.

> **NOTE**: There are two XXX lines that need to be addressed at this spot.
> 
> * [ ]  `/faq.md#submit` FAQ 0.0 - How to register
> 
> When the critical path fix (see way above) is in place and the procedure how to "Register for the IOCCC" is known, update the XXX comment.
> 
> * [ ]  `/faq.md#submit` FAQ 0.0 - How to Upload your entry
> 
> When the critical path fix (see way above) is in place and the procedure how to "How to Upload your entry" is known, update the XXX comment.
> 
Are these different from above? I mean how does it differ in what you're getting at here?

> * [ ]  `/faq.md#submit` FAQ 0.0 - change of repo
> 
> When the **great fork merge** is performed, references to _temp-test-ioccc_ should be changed to _winners_ and then the XXX note should be removed.

We can use my `sgit(1)` tool for that of course. That'll make it very easy to do.

> 
> * [ ]  `/faq.md#submit` FAQ 1.2 - great fork merge
> 
> When the **great fork merge** is performed, the XXX naked date should ne added and the XXX note should be removed.

Please clarify what you mean here. I guess the typos and being too tired to work it out are making it hard to follow.

**Comment by lcn2** [comment]
Updated TODO list.

**Comment by xexyl** [comment]
> Updated TODO list.

Thanks .. and thanks for assigning it to me.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
With commit f482380661b88ca3e0f4a70c4d9b4d5ccb17e9f0 the TODO items in this issue #2008 has been completed or moved to issue #2239 because they related to final steps of the **Great Fork Merge**.
